File: .pub-cache/hosted/pub.dartlang.org/watcher-1.0.2/test/ready/linux_test.dart
// Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

@TestOn('linux')

import 'package:test/test.dart';
import 'package:watcher/src/directory_watcher/linux.dart';

import 'shared.dart';
import '../utils.dart';

void main() {
  watcherFactory = (dir) => LinuxDirectoryWatcher(dir);

  sharedTests();
}

File: engine/src/flutter/shell/platform/linux/testing/linux_test.cc
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "flutter/shell/platform/linux/testing/linux_test.h"

#include "flutter/shell/platform/linux/fl_engine_private.h"
#include "flutter/shell/platform/linux/testing/fl_test_gtk_logs.h"

namespace flutter {
namespace testing {

LinuxTest::LinuxTest() {
  fl_ensure_gtk_init();
  loop = g_main_loop_new(nullptr, FALSE);
  project = fl_dart_project_new();
  engine = fl_engine_new(project);
}

LinuxTest::~LinuxTest() {
  g_clear_object(&project);
  g_clear_pointer(&loop, g_main_loop_unref);
}

void LinuxTest::TearDown() {
  g_clear_object(&engine);
}

void LinuxTest::StartEngine(FlEngine* engine) {
  g_autoptr(GError) error = nullptr;
  EXPECT_TRUE(fl_engine_start(engine, &error));
  EXPECT_EQ(error, nullptr);
}

}  // namespace testing
}  // namespace flutter
